### PR TITLE
Initial API Testing content

### PR DESCRIPTION
**Pull request by ericrommel**

### CODE REVIEW

File: topics/06_api_testing/main_api_testing.md [hunk 1]
```diff
@@ -1,3 +1,231 @@
...
+    response = requests.get("https://api.example.com/nonexistent")
+    response.raise_for_status()  # Will raise an error for 4xx/5xx responses
+except requests.exceptions.RequestException as e:
+    print(f"An error occurred: {e}")
+```
+
+
+## Advanced Testing Techniques - Mocking API Responses
+
+Mocking API responses can control the testing environment without making actual requests. Libraries like responses allow for mocking in Python.
```
Comment (ksauhen) [inline topics/06_api_testing/main_api_testing.md:214]: We can add a link for self-study https://realpython.com/python-mock-library/#what-is-mocking

File: topics/06_api_testing/main_api_testing.md [hunk 2]
```diff
@@ -1,3 +1,231 @@
...
+
+
+### Example
+
+```python
+import requests
+from unittest.mock import patch
+
+@patch('requests.get')
+def test_mocked_get_request(mock_get):
```
Comment (ksauhen) [inline topics/06_api_testing/main_api_testing.md:224]: The example i snot clear without comments what is going in here

File: topics/06_api_testing/main_api_testing.md [hunk 3]
```diff
@@ -1,3 +1,231 @@
 # Topic 6: API Testing with Python
 
-Coming soon.
+API Testing is a crucial component in test automation, ensuring that interactions between software components or systems meet specified requirements. This module will cover Python's tools and libraries for API testing, emphasizing HTTP requests, response validation, and creating automated API test suites.
+
+
+## Learning Objectives
```
Comment (ksauhen) [inline topics/06_api_testing/main_api_testing.md:6]: There are no references to sample folder

File: topics/06_api_testing/main_api_testing.md (hunk 2 above)
Comment (ericrommel) [reply topics/06_api_testing/main_api_testing.md]: Added more detailed explanation about it and also using fixtures and a small comparison between them.

File: topics/06_api_testing/main_api_testing.md (hunk 3 above)
Comment (ericrommel) [reply topics/06_api_testing/main_api_testing.md:6]: Added.

File: topics/06_api_testing/main_api_testing.md [hunk 4]
```diff
@@ -1,3 +1,483 @@
...
+    "completed": True
+}
+
+# Validate response data against schema
+def test_response_schema():
+    validate(instance=response_data, schema=schema)
+```
+
+
+### Testing Asynchronous APIs
```
Comment (ksauhen) [inline topics/06_api_testing/main_api_testing.md:326]: @ericrommel Asynchronous topic is for senior level specialists. Do you real want to mention it in this course?

File: topics/06_api_testing/main_api_testing.md [hunk 5]
```diff
@@ -1,3 +1,483 @@
...
+])
+def test_user_api(user_id, expected_status):
+    url = "https://jsonplaceholder.typicode.com/users"
+    params = {"id": user_id} if user_id else {}
+    response = requests.get(url, params=params)
+    assert response.status_code == expected_status
+```
+
+
+### Dynamic Request Manipulation
```
Comment (ksauhen) [inline topics/06_api_testing/main_api_testing.md:416]: Sorry, but I did not get it. What is it?

**Review by ksauhen — APPROVED** [review]

**Pull request merged**